Formula/libkeccak.rb

```diff
@@ -0,0 +1,36 @@
+class Libkeccak < Formula
+  desc "Keccak-family hashing library"
+  homepage "https://github.com/maandree/libkeccak"
+  url "https://github.com/maandree/libkeccak/archive/1.1.4.tar.gz"
+  sha256 "40e27983ea63faf47071402b212da48101b1dae37ce2702da947361ee8d4a313"
+
+  depends_on "binutils" => :build
+  depends_on "coreutils" => :build
+  depends_on "make" => :build
+
+  def install
+    system "make", "LIBEXT=dylib", "LIBFLAGS=-dynamiclib"
+    system "make", "install", "LIBEXT=dylib", "LIBFLAGS=-dynamiclib", "PREFIX=#{prefix}"
+    fix_libkeccak_symlinks
+  end
+
+  def fix_libkeccak_symlinks
+    mv "#{lib}/libkeccak.dylib.1.1", "#{lib}/libkeccak.1.1.dylib"
+    ln_s "#{lib}/libkeccak.1.1.dylib", "#{lib}/libkeccak.1.dylib"
+    rm "#{lib}/libkeccak.dylib"
+    ln_s "#{lib}/libkeccak.1.1.dylib", "#{lib}/libkeccak.dylib"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <libkeccak.h>
+      int main() {
+      libkeccak_spec_t spec;
+        libkeccak_state_t state;
+        return 0;
+      }
```

Comment:
We need a better test here. It looks like there are two files `test.c` and `.testfile` that could be used for more robust testing. You could put these in a directory in the source tree (e.g., `./examples`) and then use `pkgshare.install "examples"` during installation. Then for your test you could validate successful installation by building and running `test.c` and validating it's output.

A formula like `z3` might be a good place to look for inspiration (`brew cat z3`).